dtfs-central-api/src/services/state-machines/fee-record/event-handlers/helpers/calculate-fixed-fee.ts

```diff
@@ -8,19 +8,16 @@ import { getDateFromMonthAndYear, MonthAndYear, ReportPeriod } from '@ukef/dtfs2
  */
 const BANK_ADMIN_FEE_ADJUSTMENT = 0.9;
 
-const getNumberOfDaysRemainingInCoverPeriod = (reportPeriodStart: MonthAndYear, coverStartDate: Date, coverEndDate: Date): number => {
-  const currentReportPeriodStartMonthStart = startOfMonth(getDateFromMonthAndYear(reportPeriodStart));
+const getNumberOfDaysRemainingInCoverPeriod = (reportPeriodEnd: MonthAndYear, coverEndDate: Date): number => {
+  const endDateOfReportPeriod = startOfDay(endOfMonth(getDateFromMonthAndYear(reportPeriodEnd)));
```

Comment:
I'd like to get people's thoughts here on readability. 
endOfMonth will give something like 2024-10-31:23:59:59 and differenceInDays only does whole day difference, so if I want to find the number of days between the 1st day after the reporting period and cover end date I can either do:
```
const endDate = endOfMonth(getDateFromMonthAndYear(reportPeriodEnd))
return differenceInDays(coverEndDate, startDateOfNextReportPeriod);
```

Or the explicit startOfDay then adding a day style thing I've done here to show exactly what date I care about starting from (and not obscuring the kind of 23:59:59 behaviour of endOfMonth)

More code but maybe more readable?

Reply:
I think I prefer the `startOfDay` and then adding a day as you have done as otherwise I would have read it as being a day out. 

That said, in either case it might be good to leave a comment in the code since I think it is unlikely that all future maintainers will be fully aware of the specifics of this?

Reply:
Good shout, have added documentation to say it's calculating day after report period end to end of cover date (think that's enough without a bunch of info on how date-fns works)